tools/crm_mon.c

```diff
@@ -1953,8 +1956,8 @@ mon_refresh_display(gpointer user_data)
 
         case mon_output_legacy_xml:
         case mon_output_xml:
-            print_xml_status(out, mon_data_set, stonith_history, options.mon_ops,
-                             show, options.neg_location_prefix);
+            print_xml_status(out, mon_data_set, history_rc, stonith_history,
+                             options.mon_ops, show, options.neg_location_prefix);
```

Comment:
history_rc is a legacy return code, but even if it were mapped to a standard one, it can be a system errno, which can vary by OS and so can not be reliably interpreted. I'm thinking we should use crm_errno2exit(history_rc) since crm_exit_t is identically defined across all platforms. It's also consistent with the overall status, so callers only have to care about one set of codes. callers can use crm_error --exit to get a readable version.

In practice the currently possible errors are -ENOTCONN (-> CRM_EX_DISCONNECT), -EINVAL (-> CRM_EX_SOFTWARE), -ECOMM (-> CRM_EX_OSERR), and -ENOMSG (-> CRM_EX_PROTOCOL).

Reply:
Okay, I think I've got this taken care of.